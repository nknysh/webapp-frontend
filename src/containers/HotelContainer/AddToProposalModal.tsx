import React, { useState, useEffect } from 'react';

import { PrimaryButton } from 'pureUi/Buttons';
import Label from 'pureUi/Label';
import Select from 'pureUi/Select';
import Input from 'pureUi/Input';

import { useTranslation } from 'react-i18next';
import { ValueLabelPair, ReduxDomainStatus } from '../../interfaces';

export const AddToProposalModalContent = props => {
  const { t } = useTranslation();

  const { createNewProposal, addToProposal }: { createNewProposal: Function; addToProposal: Function } = props;
  const { proposalStatus }: { proposalStatus: ReduxDomainStatus } = props;
  const { proposalResult, hotelUuid, history }: { proposalResult: string; hotelUuid: string; history: any } = props;
  const { proposals }: { proposals: any } = props;

  const [selectedProposalUuid, setSelectedProposalUuid] = useState('new');
  const [newProposalName, setNewProposalName] = useState('');
  const [isNewProposal, setIsNewProposal] = useState(selectedProposalUuid === 'new');
  const [isSubmitted, setIsSubmitted] = useState(false);

  // when the modal is submitted and the proposal status is SUCCESS, we're complete
  useEffect(() => {
    if (isSubmitted && proposalStatus === 'SUCCESS') {
      history.push(`/proposals/${proposalResult}/edit`);
    }
  }, [isSubmitted, proposalStatus]);

  const handleProposalNameChange = e => {
    if (e.target.value === 'new') {
      setIsNewProposal(true);
    } else {
      setIsNewProposal(false);
    }
    setSelectedProposalUuid(e.target.value);
  };

  const handleAddToProposalSubmit = () => {
    // if the proposal status is loading, return out - we dont want to send multiple requests
    // @see https://pureescapes.atlassian.net/browse/OWA-1067
    if (proposalStatus === 'LOADING') {
      return;
    }

    // we should NEVER place holds as we are making proposals which is why a hard `false` is passed here
    if (isNewProposal) {
      createNewProposal(newProposalName, hotelUuid, false);
    } else {
      addToProposal(selectedProposalUuid, hotelUuid, false);
    }
    setIsSubmitted(true);
  };

  const selectOptions: ValueLabelPair[] = [];

  selectOptions.push({
    value: 'new',
    label: t('labels.newProposal'),
  });

  Object.keys(proposals).map(pKey =>
    selectOptions.push({
      value: pKey,
      label: proposals[pKey],
    })
  );

  return (
    <div className="add-to-proposal">
      <Label className="mb-4">{t('labels.proposalId')}</Label>
      <Select
        className="mb-4"
        value={selectedProposalUuid}
        options={selectOptions}
        onChange={handleProposalNameChange}
      />

      {isNewProposal && (
        <div className="mb-4">
          <Label className="mb-4">{t('labels.proposalName')}</Label>
          <Input value={newProposalName} onChange={e => setNewProposalName(e.target.value)} />
        </div>
      )}

      <PrimaryButton type="button" onClick={handleAddToProposalSubmit}>
        {t('buttons.addToProposal')}
      </PrimaryButton>
    </div>
  );
};
