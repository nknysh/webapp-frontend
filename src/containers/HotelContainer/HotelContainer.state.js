import { connect } from 'react-redux';
import { pipe } from 'ramda';

import { getHotel, getHotelsBrochures, getHotelsPhotos } from 'store/modules/hotels';
import { fetchHotel } from 'store/modules/hotel';
import { addRoom, getBooking, removeRoom, updateBooking } from 'store/modules/bookings';

import {
  bookingCanBookSelector,
  bookingCanHoldSelector,
  bookingPaymentTermsSelector,
  bookingCancellationPoliciesSelector,
  bookingOffersTermsSelector,
} from 'store/modules/fastSearch';
import { initializeBookingBuilderAction } from 'store/modules/bookingBuilder/actions';

export const mapStateToProps = (state, { id }) => ({
  booking: getBooking(state, id),
  brochures: getHotelsBrochures(state, id),
  canBook: bookingCanBookSelector(state),
  canHold: bookingCanHoldSelector(state),
  hotel: getHotel(state, id),
  photos: getHotelsPhotos(state, id),
  paymentTerms: bookingPaymentTermsSelector(state),
  cancellationPolicy: bookingCancellationPoliciesSelector(state),
  offersTerms: bookingOffersTermsSelector(state),
});

export const mapDispatchToProps = dispatch => ({
  fetchHotel: pipe(
    fetchHotel,
    dispatch
  ),
  addRoom: pipe(
    addRoom,
    dispatch
  ),
  removeRoom: pipe(
    removeRoom,
    dispatch
  ),
  updateBooking: pipe(
    updateBooking,
    dispatch
  ),
  initializeBooking: pipe(
    initializeBookingBuilderAction,
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
);
