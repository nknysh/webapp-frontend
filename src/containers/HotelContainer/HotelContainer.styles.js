import styled from 'styled-components';
import { Container, BackButton, Modal, Button } from '@pure-escapes/webapp-ui-components';

import { Breadcrumbs, Hotel } from 'components';
import SummaryForm from 'containers/SummaryForm';

import { theme } from 'styles';

export const StyledHotelContainer = styled(Container)`
  margin-bottom: ${theme.spacing.gutter * 2}px;

  ${props => props.theme.breakpoints.tablet`
    padding-top: ${theme.spacing.gutter * 6.5}px;
  `}
`;

export const StyledHotel = styled(Hotel)`
  ${props => props.theme.breakpoints.tablet`
        flex: 1;
        max-width: 64%;
        padding: 0;
        margin-top: ${theme.spacing.gutter * 2}px;
    `}
`;

export const StyledSummary = styled(SummaryForm)``;

export const StyledBreadcrumbs = styled(Breadcrumbs)`
  margin-left: ${theme.spacing.gutter}px;
  margin-right: ${theme.spacing.gutter}px;
`;

export const Full = styled.div`
  display: flex;
`;

export const Back = styled(BackButton)`
  display: flex;
  padding ${theme.spacing.gutter * 2}px !important;
`;

export const Aside = styled.aside`
  position: relative;

  ${props => props.theme.breakpoints.tablet`
    flex: 1;
    max-width: 36%;
    margin-top: ${theme.spacing.gutter * 2}px;
  `}
`;

export const Title = styled.h3`
  font-size: ${theme.fonts.sizes.default}px;
  padding: 0 0 ${theme.spacing.gutter * 1.45}px;
  text-transform: uppercase;
  font-weight: ${theme.fonts.bold};
  letter-spacing: ${theme.fonts.letterSpacing.medium}px;
  line-height: 14px;
  color: ${theme.palette.neutral};
  border-bottom: 1px solid ${theme.borders.default};
`;

export const AsideDetails = styled.div`
  margin: 0 ${theme.spacing.gutter * 2}px ${theme.spacing.gutter * 3.5}px;

  ${props => props.theme.breakpoints.tablet`
    margin-left: 0;
    margin-right: 0;
  `}
`;

export const Brochure = styled.a`
  display: block;
  cursor: pointer;
  color: ${theme.palette.primary};
  font-size: ${theme.fonts.sizes.default}px;
  font-weight: ${theme.fonts.bold};
  text-transform: uppercase;
  line-height: 14px;
  letter-spacing: 0.38px;
  padding: ${theme.spacing.gutter * 0.75}px 0;

  :before {
    content: '+ ';
  }
`;

export const StyledModal = styled(Modal)``;

export const SummaryActions = styled.div`
  display: flex;
  margin: ${theme.spacing.gutter * 1.5}px 0;
`;

export const SummaryAction = styled(Button)`
  background: ${theme.palette.light};
  flex: 1 1 50%;
  margin: 0 ${theme.spacing.gutter}px;

  :first-child {
    margin-left: 0;
  }

  :last-child {
    margin-right: 0;
  }
`;

export const Text = styled.p`
  color: ${theme.palette.secondary};
  font-size: ${theme.fonts.sizes.normal}px;
`;
