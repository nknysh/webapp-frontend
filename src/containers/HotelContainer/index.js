export { default as HotelContainer } from './HotelContainer';
