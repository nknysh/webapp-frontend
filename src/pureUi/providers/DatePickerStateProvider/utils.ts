import { format, isSameMonth, isSameYear } from 'date-fns';
import { formatDateRangeDisplay } from 'utils/date';

export const getDateRangeDisplayStringDeprecated = (startDate: string, endDate: string): string => {
  if (!endDate) {
    return startDate ? format(new Date(startDate), 'LLL') : 'Select date range';
  }

  const startDateObj = new Date(startDate);
  const endDateObj = new Date(endDate);
  const startDay = format(startDateObj, 'd');
  const endDay = format(endDateObj, 'd');
  const startMonth = format(startDateObj, 'LLL');
  const endMonth = format(endDateObj, 'LLL');
  const startYear = format(startDateObj, 'yyyy');
  const endYear = format(endDateObj, 'yyyy');
  const inSameMonth = isSameMonth(startDateObj, endDateObj);
  const inSameYear = isSameYear(startDateObj, endDateObj);

  if (startDate === endDate) {
    return `${startDay} ${startMonth} ${startYear}`;
  }

  if (!inSameMonth && inSameYear) {
    return `${startDay} ${startMonth} - ${endDay} ${endMonth} ${endYear}`;
  }

  if (!inSameMonth && !inSameYear) {
    return `${startDay} ${startMonth} ${startYear} - ${endDay} ${endMonth} ${endYear}`;
  }

  //  Default same month/year
  return `${startDay} - ${endDay} ${startMonth} ${endYear}`;
};

export const getDateRangeDisplayString = (startDate: string, endDate: string): string => {
  return formatDateRangeDisplay(startDate, endDate) || 'Select date range';
};
