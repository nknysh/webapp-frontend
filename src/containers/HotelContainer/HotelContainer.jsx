import React, { Fragment } from 'react';
import { __, pipe, lensProp, set, compose, prop, path, allPass, has, complement, curry, view } from 'ramda';

import { Loader, Tabs } from 'components';
import { BookingContainer } from 'containers';
import { useFetchData, useCurrentWidth } from 'effects';
import { isMobile } from 'utils';

import uiConfig from 'config/ui';

import connect from './HotelContainer.state';
import { propTypes, defaultProps } from './HotelContainer.props';
import {
  StyledHotelContainer,
  Back,
  StyledBreadcrumbs,
  StyledHotel,
  StyledSummary,
  Full,
} from './HotelContainer.styles';

const roomsLens = lensProp('rooms');

const reloadIfMissing = complement(allPass([has('photos'), has('rooms')]));

const renderBackButton = () => <Back to="/search">{path(['labels', 'backToSearch'], uiConfig)}</Back>;

export const HotelContainer = ({ hotel, id, fetchHotel, hotelStatus, getBooking, updateBooking }) => {
  const loaded = useFetchData(hotelStatus, fetchHotel, id, reloadIfMissing(hotel));
  const currentWidth = useCurrentWidth();

  const booking = getBooking(id);

  const setBooking = set(__, __, booking);
  const viewBooking = view(__, booking);

  const setSelectedRooms = pipe(
    setBooking(roomsLens),
    curry(updateBooking)(id)
  );

  const renderBreadcrumbs = () => (
    <StyledBreadcrumbs links={[{ label: renderBackButton() }, { label: prop('name', hotel), to: `/hotels/${id}` }]} />
  );

  const renderHotel = () => (
    <StyledHotel onRoomSelect={setSelectedRooms} selectedRooms={viewBooking(roomsLens)} {...hotel} />
  );

  const renderSummary = () => <BookingContainer Component={StyledSummary} hotelUuid={id} />;

  const renderTabs = () => (
    <Fragment>
      {renderBackButton()}
      <Tabs centered labels={[path(['labels', 'hotelDetails'], uiConfig), path(['labels', 'yourSelection'], uiConfig)]}>
        {renderHotel()}
        {renderSummary()}
      </Tabs>
    </Fragment>
  );

  const renderFull = () => (
    <Fragment>
      {renderBreadcrumbs()}
      <Full>
        {renderHotel()}
        {renderSummary()}
      </Full>
    </Fragment>
  );

  return (
    <Loader isLoading={!loaded} text={path(['messages', 'gettingHotel'], uiConfig)}>
      <StyledHotelContainer>{isMobile(currentWidth) ? renderTabs() : renderFull()}</StyledHotelContainer>
    </Loader>
  );
};

HotelContainer.propTypes = propTypes;
HotelContainer.defaultProps = defaultProps;

export default compose(connect)(HotelContainer);
