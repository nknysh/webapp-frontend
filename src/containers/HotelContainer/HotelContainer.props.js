import PropTypes from 'prop-types';

import { noop } from 'utils';

export const propTypes = {
  hotel: PropTypes.object,
  policiesAndTerms: PropTypes.shape({
    paymentTerms: PropTypes.object,
    cancellationPolicy: PropTypes.object,
  }),
  updateBooking: PropTypes.func,
  resetBookingBuilderUiStateAction: PropTypes.func,
};

export const defaultProps = {
  hotel: {},
  updateBooking: noop,
  policiesAndTerms: {
    paymentTerms: {},
    cancellationPolicy: {},
  },
};
