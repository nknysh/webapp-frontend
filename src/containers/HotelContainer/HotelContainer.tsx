import React, { Fragment, useState, Children, useCallback, useEffect } from 'react';
import { withRouter, Redirect } from 'react-router-dom';
import { compose, isEmpty, map, prop, values } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import { useTranslation } from 'react-i18next';
import { Loader, Tabs, List } from '@pure-escapes/webapp-ui-components';
import { withUser } from 'hoc';
import { useCurrentWidth } from 'effects';
import { mapWithIndex } from 'utils';
import { StandardModal, ModalContent, ModalHeader } from 'pureUi/Modal';
import { Icon } from '@material-ui/core';
import styled from 'styled-components';
import { pureUiTheme } from 'pureUi/pureUiTheme';
import { Heading1 } from 'styles';

import connect from './HotelContainer.state';
import { propTypes, defaultProps } from './HotelContainer.props';
import {
  AsideDetails,
  Back,
  Full,
  StyledBreadcrumbs,
  StyledHotelContainer,
  Text,
  Title,
} from './HotelContainer.styles';
import { Hotel } from 'components';
import SummaryForm from 'containers/SummaryForm';
import { makeBackendApi } from 'services/BackendApi';
import { IReduxDomainStatus } from '../../interfaces';
import { AddToProposalModalContent } from './AddToProposalModal';
import {
  TableCardBox,
  TableCardRow,
  TableCardNumberedBanner,
  TableCardNumberBannerNumber,
  TableCardNumberBannerText,
} from 'pureUi/TableCard';
const LeftColumn = props => {
  const { id, hotel, photos } = props;

  return (
    <div
      style={{
        flex: 1,
        paddingLeft: '14px',
        paddingRight: '14px',
      }}
    >
      <TableCardNumberedBanner className="mb-4">
        <TableCardNumberBannerNumber>1</TableCardNumberBannerNumber>
        <TableCardNumberBannerText>Select Room Type</TableCardNumberBannerText>
      </TableCardNumberedBanner>
      <Hotel {...hotel} id={id} photos={photos} />
    </div>
  );
};

const RightColumn = props => {
  return (
    <div
      style={{
        flex: 1,
        paddingLeft: '14px',
        paddingRight: '14px',
      }}
    >
      <HotelSummary {...props} />
    </div>
  );
};

const renderBackButton = t => <Back to="/search/beta">{t('labels.backToSearch')}</Back>;

const renderBrochure = ({ uuid, displayName, url }) => (
  <TableCardRow key={uuid} depth={3} className="table-card-row brochure-row">
    <span>{displayName}</span>
    <a href={url} target="_blank">
      <Icon>get_app</Icon>
    </a>
  </TableCardRow>
);

const HotelSummary = props => {
  const { proposals, history, booking, hotel, paymentTerms, cancellationPolicy, offersTerms, t, id, brochures, actingCountryCode } = props;
  const { proposalResult, proposalStatus }: { proposalResult: string; proposalStatus: IReduxDomainStatus } = props;
  const { canBook, canHold, onTakeHold }: { canBook: boolean; canHold: boolean; onTakeHold: boolean } = props;
  const {
    fetchProposals,
    createNewProposal,
    addToProposal,
  }: { fetchProposals: Function; createNewProposal: Function; addToProposal: Function } = props;

  const backendApi = makeBackendApi(actingCountryCode);

  const [isModalOpen, setIsModalOpen] = useState(false);

  // we ALWAYS need to fetch the new proposals
  useEffect(() => {
    async function load() {
      await fetchProposals();
    }
    load();
  }, [fetchProposals]);

  const handleAddToProposalClick = e => {
    e.preventDefault();
    setIsModalOpen(true);
  };

  const handleCreateNewProposalClick = (name, bookingId, placeHolds) => {
    createNewProposal(name, bookingId, placeHolds, backendApi);
  };

  return (
    <aside id="aside">
      <TableCardNumberedBanner className="mb-4">
        <TableCardNumberBannerNumber>2</TableCardNumberBannerNumber>
        <TableCardNumberBannerText>Review Selections</TableCardNumberBannerText>
      </TableCardNumberedBanner>
      {isModalOpen && (
        <StandardModal onClose={() => setIsModalOpen(false)}>
          <ModalHeader>
            <Heading1>{t('buttons.addToProposal')}</Heading1>
          </ModalHeader>
          <ModalContent>
            <AddToProposalModalContent
              proposals={proposals}
              hotelUuid={booking && booking.request && booking.request.hotelUuid ? booking.request.hotelUuid : null}
              createNewProposal={handleCreateNewProposalClick}
              addToProposal={addToProposal}
              proposalStatus={proposalStatus}
              proposalResult={proposalResult}
              history={history}
            />
          </ModalContent>
        </StandardModal>
      )}
      <SummaryForm
        id={id}
        showRoomImage={false}
        canHold={canHold}
        handleAddToProposalClick={handleAddToProposalClick}
      />

      {hotel.additionalInfo && (
        <TableCardBox className="mt-4">
          <TableCardRow depth={3} hasOriginalLineHeight={true}>
            <Title>{t('labels.thingsToBeAwareOf')}</Title>
            <Text>{hotel.additionalInfo}</Text>
          </TableCardRow>
        </TableCardBox>
      )}
      {!isNilOrEmpty(hotel.policiesAndRestrictions) && (
        <TableCardBox className="mt-4">
          <TableCardRow depth={3} hasOriginalLineHeight={true}>
            <Title>{t('labels.policiesAndRestrictions')}</Title>
            <List>{Children.toArray(hotel.policiesAndRestrictions)}</List>
          </TableCardRow>
        </TableCardBox>
      )}
      {!isNilOrEmpty(cancellationPolicy) && (
        <TableCardBox className="mt-4">
          <TableCardRow depth={3} hasOriginalLineHeight={true}>
            <Title>{t('labels.cancellationPolicy')}</Title>
            <List>{cancellationPolicy}</List>
          </TableCardRow>
        </TableCardBox>
      )}
      {!isNilOrEmpty(paymentTerms) && (
        <TableCardBox className="mt-4">
          <TableCardRow depth={3} hasOriginalLineHeight={true}>
            <Title>{t('labels.paymentTerms')}</Title>
            <List>{paymentTerms}</List>
          </TableCardRow>
        </TableCardBox>
      )}
      {!isNilOrEmpty(offersTerms) && (
        <TableCardBox className="mt-4">
          <TableCardRow depth={3} hasOriginalLineHeight={true}>
            <AsideDetails>
              <Title>{t('labels.offersTerms')}</Title>
              <List>
                {mapWithIndex(
                  ({ name, termsAndConditions }, i) => (
                    <Fragment key={i}>
                      <span>{name}</span>
                      <p>{termsAndConditions}</p>
                    </Fragment>
                  ),
                  values(offersTerms)
                )}
              </List>
            </AsideDetails>
          </TableCardRow>
        </TableCardBox>
      )}
      {!isEmpty(brochures) && (
        <TableCardBox className="mt-4">
          <AsideDetails>
            <TableCardRow depth={3} hasOriginalLineHeight={true}>
              <Title>{t('brochure_plural')}</Title>
            </TableCardRow>
            {values(map(renderBrochure, brochures))}
          </AsideDetails>
        </TableCardBox>
      )}
    </aside>
  );
};

export const HotelFullLayout = props => {
  const { t, id, hotel, photos } = props;

  return (
    <Fragment>
      <StyledBreadcrumbs
        links={[{ label: renderBackButton(t) }, { label: prop('name', hotel), to: `/hotels/${id}` }]}
      />
      <Full>
        <div
          style={{
            width: '50%',
          }}
        >
          <LeftColumn id={id} photos={photos} hotel={hotel} />
        </div>
        <div
          style={{
            width: '50%',
          }}
        >
          <RightColumn {...props} />
        </div>
      </Full>
    </Fragment>
  );
};

export const HotelTabLayout = props => {
  const { t, id, hotel, photos } = props;

  return (
    <Fragment>
      <StyledBreadcrumbs
        links={[{ label: renderBackButton(t) }, { label: prop('name', hotel), to: `/hotels/${id}` }]}
      />
      <Tabs labels={['Rooms', 'Booking']}>
        <div style={{ paddingTop: '14px' }}>
          <LeftColumn id={id} photos={photos} hotel={hotel} />
        </div>
        <div style={{ paddingTop: '14px' }}>
          <RightColumn {...props} />
        </div>
      </Tabs>
    </Fragment>
  );
};

export const HotelContainer = ({
  className,
  history,
  fetchHotel,
  hotel,
  photos,
  id,
  resetBookingBuilderUiState,
  ...props
}) => {
  const { t } = useTranslation();
  const [redirectToHold, setRedirectToHold] = useState(false);
  const [isLoading, setIsLoading] = useState(true);
  const { initializeBooking, match, selectedTa } = props;

  const { isMobile } = useCurrentWidth();

  useEffect(() => {
    return () => {
      resetBookingBuilderUiState();
    };
  }, []);

  useEffect(() => {
    // if we have a selectedTa from the withTravelAgent domain
    //   initialise with that too
    if (selectedTa && selectedTa.uuid) {
      initializeBooking(match.params.id, selectedTa.uuid);
    } else {
      initializeBooking(match.params.id);
    }
  }, [initializeBooking, match, selectedTa]);

  useEffect(() => {
    async function load() {
      await fetchHotel(id);
      setIsLoading(false);
    }
    load();
  }, [fetchHotel, id]);

  const onTakeHold = useCallback(() => setRedirectToHold(true), []);
  if (redirectToHold) return <Redirect to={`/hotels/${id}/hold`} />;

  const defaultProps = { history, hotel, photos, id, onTakeHold, ...props };

  const renderWithLoader = () => (
    <Loader isLoading={isLoading} text={t('messages.gettingHotel')}>
      {renderWithoutLoader()}
    </Loader>
  );

  const renderWithoutLoader = () => (
    <React.Fragment>
      <StyledHotelContainer>
        {isMobile && <HotelTabLayout t={t} {...defaultProps} />}
        {!isMobile && <HotelFullLayout t={t} {...defaultProps} />}
      </StyledHotelContainer>
    </React.Fragment>
  );

  return <div className={className}>{isEmpty(hotel) ? renderWithLoader() : renderWithoutLoader()}</div>;
};

HotelContainer.propTypes = propTypes;
HotelContainer.defaultProps = defaultProps;

const ConnectedHotel = compose(
  connect,
  // @ts-ignore
  withRouter,
  withUser
)(HotelContainer);

export default styled(ConnectedHotel)`
  .table-card-row.brochure-row {
    display: flex;
    padding-bottom: 12px;
    padding-top: 12px;
    line-height: 14px;
    padding-left: 40px;
    font-size: 12px;
    text-transform: uppercase;
    color: #736a65;
    border-bottom: 1px solid ${pureUiTheme.colors.grayDark};

    &:nth-child(2) {
      padding-top: 0px;
    }

    span {
      flex: 1;
      &::before {
        content: '';
        border-color: transparent #111;
        border-style: solid;
        border-width: 3px 0 3px 4px;
        display: block;
        height: 0;
        width: 0;
        left: -10px;
        top: 10px;
        position: relative;
      }
    }

    a {
      display: block;
      margin-top: 4px;
      margin-bottom: -4px;
      color: ${pureUiTheme.colors.gold};

      &:hover {
        color: ${pureUiTheme.colors.teal};
      }
    }
  }
`;
