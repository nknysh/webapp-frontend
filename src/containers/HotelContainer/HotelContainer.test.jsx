import React from 'react';

import { noop } from 'utils';

import { HotelContainer } from './HotelContainer';

const getComponent = props =>
  shallow(<HotelContainer getBooking={noop} brochures={{}} getHotelUploads={noop} {...props} />);

describe('<HotelContainer />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
