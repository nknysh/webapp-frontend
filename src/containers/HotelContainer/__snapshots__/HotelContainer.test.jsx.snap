// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HotelContainer /> render matches snapshot 1`] = `
<Loader
  isLoading={true}
  showSpinner={true}
  text="Getting hotel details..."
>
  <HotelContainerstyles__StyledHotelContainer>
    <HotelContainerstyles__StyledBreadcrumbs
      links={
        Array [
          Object {
            "label": <ForwardRef
              to="/search"
            >
              back to search results
            </ForwardRef>,
          },
          Object {
            "label": undefined,
            "to": "/hotels/undefined",
          },
        ]
      }
    />
    <HotelContainerstyles__Full>
      <HotelContainerstyles__StyledHotel />
      <HotelContainerstyles__Aside>
        <HotelContainerstyles__StyledSummary
          onSubmit={[Function]}
        >
          <Component />
        </HotelContainerstyles__StyledSummary>
      </HotelContainerstyles__Aside>
    </HotelContainerstyles__Full>
  </HotelContainerstyles__StyledHotelContainer>
</Loader>
`;
